apiserver/remoterelations/remoterelations.go

```diff
@@ -603,6 +627,77 @@ func (api *RemoteRelationsAPI) registerRemoteRelation(relation params.RegisterRe
 	}, nil
 }
 
+// UnregisterRemoteRelations removes the remote relations from
+// the local model. This operation is idempotent.
+func (api *RemoteRelationsAPI) UnregisterRemoteRelations(
+	relations params.RemoteEntities,
+) (params.ErrorResults, error) {
+	results := params.ErrorResults{
+		Results: make([]params.ErrorResult, len(relations.Entities)),
+	}
+	for i, relation := range relations.Entities {
+		if err := api.unregisterRemoteRelation(relation); err != nil {
+			results.Results[i].Error = common.ServerError(err)
+		}
+	}
+	return results, nil
+}
+
+func (api *RemoteRelationsAPI) unregisterRemoteRelation(relation params.RemoteEntityId) error {
+	logger.Debugf("unregister remote relation id %+v in model %v", relation, api.st.ModelUUID())
+	// TODO(wallyworld) - do this as a transaction so the result is atomic
+
+	relModelTag := names.NewModelTag(relation.ModelUUID)
+	localRelTag, err := api.st.GetRemoteEntity(relModelTag, relation.Token)
+	if errors.IsNotFound(err) {
+		logger.Debugf("%v not found, exit early", relation)
+		return nil
+	} else if err != nil {
+		return errors.Trace(err)
+	}
+
+	logger.Debugf("unregister local end of remote relation: %v", localRelTag)
+	localRel, err := api.st.KeyRelation(localRelTag.Id())
+	if errors.IsNotFound(err) {
+		logger.Debugf("%v not found, exit early", localRelTag)
+		return nil
+	} else if err != nil {
+		return errors.Trace(err)
+	}
+
+	// Find the remote application entity for the relation.
+	var remoteApp RemoteApplication
+	for _, ep := range localRel.Endpoints() {
+		remoteApp, err = api.st.RemoteApplication(ep.ApplicationName)
+		if err == nil {
+			break
+		}
+	}
+
+	// If the remote application proxy still exists, destroy it.
+	if remoteApp != nil && remoteApp.Registered() {
```

Comment:
It would mean this was the "consuming" model - the one from which the remote relation was initiated. We only want to destroy the remote application proxy on the other side, the offering side, where the remote app is obfuscated with a name like remote-<uuid>. There's a 1:1 between these remote apps on the offering side and a consuming relation. When the relation goes away, so too does the remote app.